File: Android_Interview_Prep_App/NativeAndroid/src/main/java/com/dreams/androidquizapp/models/Answer.java
package com.dreams.androidquizapp.models;

import android.os.Parcel;
import android.os.Parcelable;

/**
 * Created by Tc2r on 5/19/2017.
 * <p>
 * Description:
 */

public class Answer implements DomainObject, Parcelable {


  private Integer id;
	private String answer;
	private String details;


	@Override
  public Integer getId()
  {
    return id;
  }

  @Override
  public void setId(Integer id)
  {
    this.id = id;

  }

	public Answer() {
	}
//	public Answer(String answer, String details) {
//		this.answer = answer;
//		this.details = details;
//	}
//	public Answer(String answer) {
//		this.answer = answer;
//		this.details = "";
//	}

	public String getAnswer() {
		return answer;
	}

	public void setAnswer(String answer) {
		this.answer = answer;
	}

	public String getDetails() {
		return details;
	}

	public void setDetails(String details) {
		this.details = details;
	}


	@Override
	public int describeContents() {
		return 0;
	}
	@Override
	public void writeToParcel(Parcel dest, int flags) {
		dest.writeString(this.answer);
		dest.writeString(this.details);
	}
	protected Answer(Parcel in) {
		this.answer = in.readString();
		this.details = in.readString();
	}

	public static final Parcelable.Creator<Answer> CREATOR = new Parcelable.Creator<Answer>() {
		@Override
		public Answer createFromParcel(Parcel source) {
			return new Answer(source);
		}

		@Override
		public Answer[] newArray(int size) {
			return new Answer[size];
		}
	};

}

File: Interview_Prep_App/native_kotlin/base/src/main/java/com/nudennie/interviewprepapp/models/Answer.kt
package com.nudennie.interviewprepapp.models

import android.os.Parcel
import android.os.Parcelable
import android.os.Parcelable.Creator

/**
 * Created by Tc2r on 5/19/2017.
 *
 *
 * Description:
 */
class Answer : DomainObject, Parcelable {
    private var id: Int? = null

    // 	public Answer(String answer, String details) {
    // 		this.answer = answer;
    // 		this.details = details;
    // 	}
    // 	public Answer(String answer) {
    // 		this.answer = answer;
    // 		this.details = "";
    // 	}
    var answer: String? = null
    var details: String? = null
    override fun getId(): Int {
        return id!!
    }

    override fun setId(id: Int?) {
        this.id = id
    }

    constructor()

    override fun describeContents(): Int {
        return 0
    }

    override fun writeToParcel(dest: Parcel, flags: Int) {
        dest.writeString(answer)
        dest.writeString(details)
    }

    protected constructor(`in`: Parcel) {
        answer = `in`.readString()
        details = `in`.readString()
    }

    companion object {
        @JvmField
        val CREATOR: Creator<Answer?> = object : Creator<Answer?> {
            override fun createFromParcel(source: Parcel): Answer {
                return Answer(source)
            }

            override fun newArray(size: Int): Array<Answer?> {
                return arrayOfNulls(size)
            }
        }
    }
}
